source/extensions/filters/http/adaptive_concurrency/concurrency_controller/gradient_controller.h

```diff
@@ -0,0 +1,150 @@
+#pragma once
+
+#include <chrono>
+#include <vector>
+
+#include "envoy/config/filter/http/adaptive_concurrency/v2alpha/adaptive_concurrency.pb.h"
+#include "envoy/config/filter/http/adaptive_concurrency/v2alpha/adaptive_concurrency.pb.validate.h"
+#include "envoy/event/dispatcher.h"
+#include "envoy/runtime/runtime.h"
+#include "envoy/stats/stats_macros.h"
+
+#include "extensions/filters/http/adaptive_concurrency/concurrency_controller/concurrency_controller.h"
+
+#include "absl/base/thread_annotations.h"
+#include "absl/synchronization/mutex.h"
+#include "circllhist.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace AdaptiveConcurrency {
+namespace ConcurrencyController {
+
+/**
+ * All stats for the gradient controller.
+ */
+// clang-format off
+#define ALL_GRADIENT_CONTROLLER_STATS(COUNTER, GAUGE) \
+  GAUGE(concurrency_limit, Accumulate)  \
```

Comment:
I don't think Accumulate makes sense for any of these? Probably NoImport for most all of them besides maybe rq_outstanding (but not sure what that means yet)?